File: 217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> uniq;
        for(auto &x: nums){
            if(uniq.find(x) != uniq.end()){
                return true;
            }
            uniq.insert(x);
        }
        return false;
    }
};

File: 217-contains-duplicate/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    if(nums.length < 2) return false;
    // with sort
    nums.sort((a,b) => a-b);
    for(let i=0; i < nums.length; i++){
        if(i > 0 && nums[i] == nums[i-1]) return true;
    }
    return false;
    // with set
    // let set = new Set();
    // for(x of nums){
    //     if(set.has(x)){
    //         return true;
    //     }
    //     set.add(x);
    // }
    // return false;
};